File: src/main/java/com/github/zametki/component/form/InputField.java
package com.github.zametki.component.form;

import com.github.zametki.util.TextUtils;
import org.apache.wicket.markup.html.form.TextField;
import org.apache.wicket.model.Model;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class InputField extends TextField<String> {
    public InputField(@NotNull String id) {
        this(id, "");
    }

    public InputField(@NotNull String id, @Nullable String val) {
        super(id, Model.of(val == null ? "" : val));
        setOutputMarkupId(true);
    }

    @NotNull
    public String getInputString() {
        return TextUtils.notNull(getModelObject(), "");
    }
}

File: src/main/java/com/github/zametki/component/form/InputField.kt
package com.github.zametki.component.form

import com.github.zametki.util.TextUtils
import org.apache.wicket.markup.html.form.TextField
import org.apache.wicket.model.Model

class InputField @JvmOverloads constructor(id: String, t: String? = "") : TextField<String>(id, Model.of(t ?: "")) {

    val inputString: String
        get() = TextUtils.notNull(modelObject, "")

    init {
        outputMarkupId = true
    }

    fun clear() {
        defaultModelObject = ""
    }
}
